[Elton Goh] iP

EltonGohJH opened the pull request:
### **Duke**

> “Object-oriented programming is an exceptionally bad idea which could only have originated in California.”  — Edsger W. Dijkstra

Duke is your ultimate task tracker. It is 
- Easy to use
- Fast
- Easy to install

To use:
1. Download it [here.](https://github.com/EltonGohJH/ip/releases/tag/A-Jar)
2. Double click it.
3. **Start using!**  😀 

Features:
- [x] Manage your task
- [ ] GUI (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        System.out.println("Life is good!");
    }
}
```

EltonGohJH committed:
Complete Level-1

EltonGohJH committed:
Complete level 2

EltonGohJH committed:
Complete level 3

EltonGohJH committed:
Complete level 4

EltonGohJH committed:
Complete A-TextUiTesting

EltonGohJH committed:
Complete level 5 and refactor some parts of the code

EltonGohJH committed:
Complete level 6 and fix some minor styling issues

EltonGohJH committed:
Complete Level-7

EltonGohJH committed:
Merge branch 'branch-Level-7'

EltonGohJH committed:
Complete Level-8

EltonGohJH committed:
Resolve merge conflicts

EltonGohJH committed:
Correct the implementation of Level-7

EltonGohJH committed:
Resolve merge conflicts

EltonGohJH committed:
Resolve merge conflicts

EltonGohJH committed:
Refactor code to extract out closely related code as classes

EltonGohJH committed:
Divide classes into packages

EltonGohJH committed:
Add junit tests

EltonGohJH committed:
Add javadocs comments

EltonGohJH committed:
Merge branch 'branch-A-JavaDoc'

EltonGohJH committed:
Tweak code to make it compliant with coding standard

EltonGohJH committed:
Resolve merge conflicts

EltonGohJH committed:
Complete level-9

EltonGohJH committed:
Merge branch 'branch-Level-9'

EltonGohJH committed:
Set up gradle and add gradle test cases

EltonGohJH committed:
Merge branch 'add-gradle-support'

EltonGohJH committed:
Add checkstyle files

EltonGohJH committed:
Merge branch 'add-checkstyle-files'

EltonGohJH committed:
Complete level 10

EltonGohJH committed:
Merge branch 'branch-Level-10'

vigneshsankariyer1234567890 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,38 @@
+        StringBuilder output = new StringBuilder();
+        output = new StringBuilder("Here are the tasks in your list:");
+        output.append("\n");
+        for (int i = 0; i < taskList.getAllTasks().size(); i++) {
```
I think you could add a toString method to your TaskList class instead and refactor this part of the code! Since the task list should show an ordered list of tasks, i think it would be more meaningful to add this there instead!

vigneshsankariyer1234567890 commented on src/main/test/java/duke/parser/ParserTest.java:
```diff
@@ -0,0 +1,27 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ParserTest {
```
very comprehensive testing provided!!! well done 👍

vigneshsankariyer1234567890 reviewed (commented):
hey Elton!

Overall, I thought the project organisation was fantastic! It was so easy to see the flow of logic and personally, the coding standards are well-adhered to! I really only have one or two comments, but other than that, great job! 

All the best for tP as well!

Vignesh

hsiaotingluv commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,21 @@
+
+public class DeleteCommand extends Command {
+
+    private final int taskNo;
```
Hey! I think your codes generally look great and I really like how you have followed the coding standard. Maybe just a suggestion for the taskNo, would it be clearer if its taskNum / numTask instead? :))

hsiaotingluv commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,117 @@
+     * @throws DukeException if no date and time or description of deadline is inputted.
+     */
+    private static Deadline createDeadline(String userInput) throws DukeException {
+        String[] splitByBy = userInput.split(" /by ");
```
is there a reason why splitByBy and not simply splitBy? thinks it works the same as well, splitByBy sounds kinda funky haha

hsiaotingluv commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,117 @@
+            throw new DukeException(":( OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
+}
```
I really like how you did your OOP, separating each command into its own new class. This makes your code very neat and concise :)))

hsiaotingluv commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,91 @@
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
```
Goodjob on importing the classes explicitly!! But since java time and io are from different packages, would it be better if you separate them with an empty line?

hsiaotingluv commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,73 @@
+    /**
+     * The date of the deadline.
+     */
+    private LocalDate date;
```
hmmm you didnt do the documentation for the attributes in the other classes, why suddenly this?

hsiaotingluv commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,73 @@
+     * @param description the description of the deadline.
+     * @param date the date of the deadline
+     * @param time the time of the deadline
+     */
```
Think you didnt have any line spacing between the description and @param in your newer files too :)) might wanna standardise the documentation?

EltonGohJH committed:
Improve the looks of the UI

EltonGohJH committed:
Add assertions to some parts of the code

This is to ensure that when the code are modified later, the invariants are not violated.

Checking the tasks before and after execution of commands allows us to ensure that the
number of tasks are updated as expected. Checking the existence of the file after storage
is created allows us to confirm that a file is created.

Let's,
* check the number of tasks before and after the execution of a command
* check if a file exists after storage is initialised.

EltonGohJH committed:
Improve the code quality

This is to ensure that the code is neatly abstracted so that it will easy to understand.

Some methods do not obey SLAP so there is a need to abstract the method into more parts.

Let's,
* update the methods in parser to make sure that SLAP is obeyed.
* update the methods in storage to make sure that SLAP is obeyed.
* fix general code formatting violations.

EltonGohJH committed:
Merge pull request #1 from EltonGohJH/branch-A-Assertions

Add assertions to some parts of the code

EltonGohJH committed:
Merge branch 'master' into branch-A-CodeQuality

EltonGohJH committed:
Merge pull request #2 from EltonGohJH/branch-A-CodeQuality

Improve the code quality

EltonGohJH committed:
Add ability to update task

EltonGohJH committed:
Add shadowjar plugin to build.gradle to generate fat jar

EltonGohJH committed:
Fix alignment of GUI

EltonGohJH committed:
Add java docs

EltonGohJH committed:
Touch up on java docs

EltonGohJH committed:
Add a user guide

EltonGohJH committed:
Add images of user guide

EltonGohJH committed:
Fix bug where images are not shown

EltonGohJH committed:
Fix some errors in readme

EltonGohJH committed:
Remove extra newline at the end of some strings

EltonGohJH committed:
Add Ui picture

EltonGohJH committed:
fix some bugs
